src/utils/fromJS.js

```diff
@@ -8,6 +8,31 @@
 
 const Immutable = require("immutable");
 
+/*
+  creates an immutable map, where each of the value's
+  items are transformed into their own map.
+
+  NOTE: we guard agains `length` being a property because
+  length confuses Immutable's internal algorithm.
+*/
+function createMap(value) {
+  if (value.hasOwnProperty("length")) {
+    value.length = `${ value.length}`;
```

Comment:
No, you reset it on the new map. But you still modified the argument. 

If you do : `let a = { length : 10 }; fromJS(a);` -> a.length is now "10" (string) instead of 10 (number)